spring-grpc-core/src/main/java/org/springframework/grpc/client/InProcessGrpcChannelFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.grpc.client;
+
+import java.util.List;
+
+import io.grpc.ChannelCredentials;
+import io.grpc.Grpc;
+import io.grpc.inprocess.InProcessChannelBuilder;
+
+/**
+ * {@link GrpcChannelFactory} that creates in-process gRPC channels.
+ *
+ * @author Chris Bono
+ */
+public class InProcessGrpcChannelFactory extends DefaultGrpcChannelFactory<InProcessChannelBuilder> {
+
+	/**
+	 * Construct an in-process channel factory instance.
+	 * @param globalCustomizers the global customizers to apply to all created channels
+	 * @param interceptorsConfigurer configures the client interceptors on the created
+	 * channels
+	 */
+	public InProcessGrpcChannelFactory(List<GrpcChannelBuilderCustomizer<InProcessChannelBuilder>> globalCustomizers,
+			ClientInterceptorsConfigurer interceptorsConfigurer) {
+		super(globalCustomizers, interceptorsConfigurer);
+		// TODO ???? setVirtualTargets((p) -> p);
```

Comment:
I think it does make sense for the IPGCF to use an identity function by default for its virtual targets and will likely update this in next commit (realized I left in as a TODO as I did my self-pr-review).